app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt

```diff
@@ -306,4 +320,17 @@ class EditProfileActivity : BaseActivity(), Injectable {
         Log.w("EditProfileActivity", "failed to pick media", throwable)
         Snackbar.make(binding.avatarButton, R.string.error_media_upload_sending, Snackbar.LENGTH_LONG).show()
     }
+
+    private fun showUnsavedChangesDialog() = lifecycleScope.launch {
```

Comment:
About these wrappings and architectural problem with the save dialog:

I see in https://github.com/tuskyapp/Tusky/pull/3965 that the AlertDialog.await() is new.

What was the idea behind that?

Other (older) locations do this rather straight forward with a `dialog.show()` and direct specification of actions on the button functions.
(See for example ComposeActivity#handleCloseButton)

Reply:
This was a new thing that @nikclayton asked me to introduce.

In short, it moves from a callback-based flow, to a more linear control flow, as you now "await" in a suspend function.

Is it "better"? Debatable.
Does it read "better" I'd say yes (but this is also debatable).

An Improvement I'd do (but we'd have to discuss it) is to make it more an extension of a LifecycleScope so you can do from an Activity or Fragment (for e.g.):

Pseudo-Code:
```
fun x() = launchAlertDialog(title, message, positive, negative, neutral, etc...) { result -> if (result == button positive) { ... } }
```

The reason is, it feels a bit strange to _await_ and pass the buttons in that call. At first it makes kinda sense, you construct the alert, then await on the buttons... so you pass them there. But it could be further abstracted I'd say.

Anyway, if you don't like this, your best bet is talk to Nik ;)

Reply:
For me this approach is much more indirect (and thus harder to read). With the old one you'll all have in one place (the call to AlertDialog.Builder).

But I guess I will discuss this if I myself have a problem (code change) like this. :-)

(Can be resolved.)

Reply:
My only question about this PR: do we still feel like this launch/await flow is better, or do we want to revert to the old style (builder)?

Reply:
Fair point, as "my problem discussion" above will probably never happen.

I have the feeling that we should leave it here, as there is another location like this, but maybe remove it in future again. (Might research the background a bit.)

Reply:
Yeah, I'd like to introduce another alternative, (merely as a draft) that modifies this await() into something different.

In lieu of the current situation, I agree with @Lakoja that we should decide in a different PR. (this is suspiciously very convenient for me so I don't have to push again... but I swear it's not the reason for my opinion) ;)

Reply:
(the truth is, I am ok with AlertDialog the way it is... not everything needs to be abstracted/wrapped for the sake of ... questionable readability).